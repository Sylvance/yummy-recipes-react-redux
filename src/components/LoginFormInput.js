import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class LoginFormInput extends Component {
  static propTypes = {
    onSave: PropTypes.func.isRequired,
    email: PropTypes.string,
    password: PropTypes.string
  }

  state = {
    email: this.props.email || '',
    password: this.props.password || ''
  }

  handleSubmit = e => {
    e.preventDefault();

    const { email, password } = this.state;
    this.props.onSave(email, password)
  }

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value })
  }

  render() {
    return (
        <form name="form" onSubmit={this.handleSubmit}>
            <input type="email"
                placeholder="Enter your email"
                autoFocus="true"
                name="email"
                value={this.state.email}
                onChange={this.handleChange}
                required />
            <input type="password"
                placeholder="Enter your password"
                autoFocus="true"
                name="password"
                value={this.state.password}
                onChange={this.handleChange}
                required />
            <input type="submit" value="Login" />
        </form>
    )
  }
}
