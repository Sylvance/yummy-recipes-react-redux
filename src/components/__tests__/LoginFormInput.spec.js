// Auto-generated do not edit

/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import LoginFormInput from '../LoginFormInput';

describe('LoginFormInput test', () => {
  it('LoginFormInput should match snapshot', () => {
    const component = renderer.create(
      <LoginFormInput
        onSave={() => {}}
        email={'defaultString'}
        password={'defaultString'}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
