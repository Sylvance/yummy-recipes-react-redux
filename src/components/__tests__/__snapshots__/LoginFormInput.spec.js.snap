// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormInput test LoginFormInput should match snapshot 1`] = `
<form
  className=""
  name="form"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="email"
    >
      Email
    </label>
    <input
      autoFocus="true"
      className="form-control"
      name="email"
      onChange={[Function]}
      placeholder="Enter your email"
      required={true}
      type="email"
      value="defaultString"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="password"
    >
      Password
    </label>
    <input
      autoFocus="true"
      className="form-control"
      name="password"
      onChange={[Function]}
      placeholder="Enter your password"
      required={true}
      type="password"
      value="defaultString"
    />
  </div>
  <button
    className="btn btn-success"
    onClick={[Function]}
    type={undefined}
  >
    Login
  </button>
</form>
`;
